dev-libs/zziplib/patches/zziplib-0.13.71.patchset

```diff
@@ -0,0 +1,189 @@
+     self.assertGreater(os.path.getsize(zipfile), 10)
+   def test_10002_make_test2_zip(self):
+     """ create a test2.zip for later tests using standard 'zip'
+@@ -306,7 +306,7 @@ class ZZipTest(unittest.TestCase):
+     filename = os.path.join(tmpdir,"README")
+     filetext = self.readme()
+     self.mkfile(filename, filetext)
+-    shell("{exe} ../{zipfile} ??*.* README".format(**locals()), cwd=tmpdir)
++    shell("printf '%s\\n' ??*.* README | {exe} ../{zipfile} -@".format(**locals()), cwd=tmpdir)
+     self.assertGreater(os.path.getsize(zipfile), 10)
+   def test_10003_make_test3_zip(self):
+     """ create a test3.zip for later tests using standard 'zip'
+@@ -322,7 +322,7 @@ class ZZipTest(unittest.TestCase):
+     filename = os.path.join(tmpdir,"README")
+     filetext = self.readme()
+     self.mkfile(filename, filetext)
+-    shell("{exe} ../{zipfile} ??*.* README".format(**locals()), cwd=tmpdir)
++    shell("printf '%s\\n' ??*.* README | {exe} ../{zipfile} -@".format(**locals()), cwd=tmpdir)
+     self.assertGreater(os.path.getsize(zipfile), 10)
+   def test_10004_make_test4_zip(self):
+     """ create a test4.zip for later tests using standard 'zip'
+@@ -339,7 +339,7 @@ class ZZipTest(unittest.TestCase):
+     filename = os.path.join(tmpdir,"README")
+     filetext = self.readme()
+     self.mkfile(filename, filetext)
+-    shell("{exe} -n README ../{zipfile} ??*.* README".format(**locals()), cwd=tmpdir)
++    shell("printf '%s\\n' ??*.* README | {exe} -n README ../{zipfile} -@".format(**locals()), cwd=tmpdir)
+     self.assertGreater(os.path.getsize(zipfile), 1000000)
+   def test_10005_make_test5_zip(self):
+     """ create a test5.zip for later tests using standard 'zip'
+@@ -363,7 +363,7 @@ class ZZipTest(unittest.TestCase):
+     filename = os.path.join(tmpdir,"README")
+     filetext = self.readme()
+     self.mkfile(filename, filetext)
+-    shell("{exe} ../{zipfile} -r file* subdir* README".format(**locals()), cwd=tmpdir)
++    shell("printf '%s\\n' file* subdir* README | {exe} ../{zipfile} -r -@".format(**locals()), cwd=tmpdir)
+     self.assertGreater(os.path.getsize(zipfile), 1000000)
+   def test_10010_make_test0_dat(self):
+     """ create test.dat from test.zip with xorcopy """
+-- 
+2.28.0
+
+
+From 6fcb226529704408ba21fb9206f5cb510bad2aa5 Mon Sep 17 00:00:00 2001
+From: begasus <begasus@gmail.com>
+Date: Sun, 27 Sep 2020 09:41:53 +0000
+Subject: Fix crash
+
+
+diff --git a/test/zziptests.py b/test/zziptests.py
+index 1c2d19d..83b0345 100644
+--- a/test/zziptests.py
++++ b/test/zziptests.py
+@@ -3488,7 +3488,7 @@ class ZZipTest(unittest.TestCase):
+     self.assertFalse(run.returncode)
+     # list the ZIPfile
+     exe=self.bins("unzip-mem");
+-    run = shell("{chdir} {tmpdir} && ../{exe} -v {zipname}.zip".format(**locals()), returncodes = [0,-8])
++    run = shell("printf '%s\\n' ??*.* | {chdir} {tmpdir} && ../{exe} -v {zipname}.zip".format(**locals()), returncodes = [0,-8])
```

Comment:
@alaviss this fixes the crash when running the tests, probably needs some love :)

Reply:
Failures are dealt with in the PATCH() function, this doesn't crash but still produces an error:
```
ERROR: test_65485_list_verbose_compressed_with_directory (__main__.ZZipTest)
 verbously list a zipfile containing directories
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/sources/zziplib-0.13.71/test/zziptests.py", line 3348, in test_65485_list_verbose_compressed_with_directory
    run = shell("printf '%s\\n' ??*.* | {chdir} {tmpdir} && ../{exe} -v {zipname}.zip".format(**locals()), returncodes = [0,-8])
  File "/sources/zziplib-0.13.71/test/zziptests.py", line 102, in shell
    raise subprocess.CalledProcessError(run.returncode, sh_command, output)
subprocess.CalledProcessError: Command 'printf '%s\n' ??*.* | cd tmp.test_65485 && ../../bins/unzip-mem -v ZIPfile.zip' returned non-zero exit status 127.
```

Reply:
Test disabled in the recipe for now, this can be fixed later